apps/api/src/leaderBoard/leaderBoard.service.ts

```diff
@@ -0,0 +1,23 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { TransformUserService } from 'src/TransformUser/TransformUser.service';
+import { Repository } from 'typeorm';
+import { User, Userfront } from 'types';
+
+@Injectable()
+export class LeaderBoardService {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+    private readonly transfornService: TransformUserService,
+  ) {}
+
+  async getLeaderBoard(): Promise<Userfront[]> {
+    const users = await this.userRepository.find({ order: { level: 'DESC' } });
+    if (!users) throw new BadRequestException('users not found');
+    const res: Userfront[] = [];
+    for (let i = 0; i < users.length; i++)
+      res.push(await this.transfornService.transform(users[i]));
```

Comment:
I think this can be optimied like that. There are a few other places where you can do it that way too

```suggestion
    const res = users.map((user) => this.transfornService.transform(users[i]));
    await Promise.all(res);
```